presto-native-execution/presto_cpp/main/http/HttpClient.h

```diff
@@ -93,7 +97,7 @@ class HttpResponse {
   FOLLY_ALWAYS_INLINE size_t nextAllocationSize(uint64_t dataLength) const;
 
   const std::unique_ptr<proxygen::HTTPMessage> headers_;
-  velox::memory::MemoryPool* const pool_;
+  const std::shared_ptr<velox::memory::MemoryPool> pool_;
```

Comment:
Can we hold the reference in PrestoExchangeSource? The ownership will be more clear and we just need to hold the reference once? Also there is a chance that the memory pool has gone while we try to hold a shared reference on it in http client. Thanks!

Reply:
PrestoExchangeSource will be gone before response returns. We create ResponseHandler before we send the request so it is guaranteed we have a valid MemoryPool at the point of shared_from_this() call while constructing ResponseHandler. We also need one shared_ptr of mpool in HttpResponse because HttpResponse can outlive ResponseHandler.

Reply:
PrestoExchangeSource is guaranteed to be there as the http request will hold a shared reference on it. The shared ptr is not freed and let's take a reference if necessary. This also make the object lifecycle clear.